pkg/cmd/join/exec.go

```diff
@@ -79,41 +121,30 @@ func (o *Options) complete(cmd *cobra.Command, args []string) (err error) {
 }
 
 func (o *Options) validate() error {
-	if o.token == "" {
-		return fmt.Errorf("token is missing")
-	}
-	if o.values.Hub.APIServer == "" {
-		return fmt.Errorf("hub-server is missing")
-	}
-	if o.values.ClusterName == "" {
-		return fmt.Errorf("name is missing")
-	}
-	if len(o.registry) == 0 {
-		return fmt.Errorf("registry should not be empty")
+	// preflight check
+	if err := preflightinterface.RunChecks(
+		[]preflightinterface.Checker{
+			preflight.HubKubeconfigCheck{
+				Config: o.HubConfig,
+			},
+			preflight.KlusterletApiserverCheck{
```

Comment:
nit: should we put "check if it is hub" and "check if it is spoke" also as preflight check. Such that this can be used in each cmd? Won't be a blocker for this pr, but worth thinking.